### PR TITLE
fix(package.json): Move ESLint related dependencies to peerDependencies section

### PR DESCRIPTION
**Pull request by Labriko**
## Description

This fix resolves compatibility issues with library consumers.

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "prepare": "husky install",
     "release": "changeset publish"
   },
-  "dependencies": {
+  "peerDependencies": {
```
Comment (smrwsky) [inline package.json:44]: Peer dependencies should be duplicated in dev dependencies so they can be installed locally.

File: package.json [hunk 2]
```diff
@@ -53,15 +53,15 @@
...
-    "prettier": "^2.8.4"
+    "eslint-plugin-react-perf": "^3.3.1"
   },
   "devDependencies": {
     "@changesets/changelog-github": "^0.4.0",
     "@changesets/cli": "^2.16.0",
     "husky": "^8.0.3",
     "license-checker": "^25.0.1",
     "lint-staged": "^13.1.2",
+    "prettier": "^2.8.4",
```
Comment (smrwsky) [inline package.json:64]: This package also provides prettier config for external usage. Shouldn't it be added in peer dependencies as well?

**Review by mvshvssv — CHANGES_REQUESTED** [review]

**Comment by mvshvssv** [comment]
Also please run changeset command and set minor update in this commit, then it will be published in npm.

File: package.json (hunk 1 above)
Comment (Labriko) [reply package.json:44]: Agree, fixed

File: package.json (hunk 2 above)
Comment (Labriko) [reply package.json:77]: Done

**Review by mvshvssv — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@justoss/code-style",
-  "version": "0.1.2",
+  "version": "0.2.0",
```
[the remark this answers is not in the document]
Comment (Labriko) [reply package.json]: Got it, thanks. Is it ok now?

**Review by mvshvssv — APPROVED** [review]

**Pull request merged**